src/assets/scripts/client/measurement/MeasureTool.js

```diff
@@ -0,0 +1,513 @@
+        const firstPoint = this._getRelativePosition(pointsList.shift());
+        const firstLeg = new MeasureLegModel(firstPoint);
+        initialValues.previousLeg = firstLeg;
+
+        pointsList.reduce((lastValues, point) => {
+            const { previousLeg } = lastValues;
+            let { totalDistance, totalDuration } = lastValues;
+            const leg = new MeasureLegModel(
+                this._getRelativePosition(point),
+                radius,
+                previousLeg
+            );
+            const distance = nm(leg.distance);
+            const bearing = heading_to_string(leg.bearing);
+            let duration = 0;
+
+            if (groundSpeed !== null) {
+                duration = Math.round((distance / groundSpeed) * TIME.ONE_HOUR_IN_SECONDS);
+            }
+
+            const labels = [
+                this._buildLabel(distance, duration, bearing)
+            ];
+
+            totalDistance += distance;
+            totalDuration += duration;
+
+            if (totalDistance !== distance) {
+                labels.push(this._buildLabel(totalDistance, totalDuration));
+            }
+
+            leg.labels = labels;
+
+            return {
+                previousLeg: leg,
+                totalDistance,
+                totalDuration
+            };
+        }, initialValues);
+
+        return {
+            initialTurn,
+            firstLeg
+        };
+    }
+
+    /**
+     * Removes the last point from the path
+     *
+     * @for MeasureTool
+     * @method removeLastPoint
+     */
+    removeLastPoint() {
+        this._throwIfNotMeasuring();
+
+        // The "last" point is actually the 2nd last item in _points
+        // This means the path will end at the cursor position when redrawn
+        const { length } = this._points;
+
+        if (length > 2) {
+            this._points.splice(length - 2, 1);
+        }
+    }
+
+    /**
+     * Updates the last point in the path
+     *
+     * The value can be an object with a `#relativePosition` property,
+     * or a array of numbers that represent a `relativePosition`
+     *
+     * @for MeasureTool
+     * @method updateEndPoint
+     * @param value {AircraftModel|FixModel|array<number>}
+     */
+    updateLastPoint(value) {
+        this._throwIfNotMeasuring();
+        this._throwIfPointInvalid(value);
+
+        if (!this.hasLegs) {
+            this.addPoint(value);
+
+            return;
+        }
+
+        this._points[this._points.length - 1] = value;
+    }
+
+    // ------------------------------ PRIVATE ------------------------------
+
+    /**
+     * Builds the initial turn parameters that are needed to fly to the first fix
+     *
+     * @for MeasureTools
+     * @method _buildInitialTurnParameters
+     * @returns {object}
+     * @private
+     */
+    _buildInitialTurnParameters() {
+        if (!this._hasStyle(MEASURE_TOOL_STYLE.INITIAL_TURN)) {
+            return null;
+        }
+
+        const { aircraft } = this;
```

Comment:
Yup, the tool takes a `FixModel`, `AircraftModel` or `relativePosition` as any point. The `aircraft` property is simply an alias for the first point (providing it's an instance of `AircraftModel`).

This could possibly be seen as a feature. If the draw style is `Arc to next fix`, starting from a fix (or cursor position) and ending in an aircraft will result in a straight line. So you can still get "as the crow flies" distance.